src/Configuration/Configuration.php

```diff
@@ -2789,51 +2832,55 @@ static function (string $annotation): ?string {
             )
         );
 
-        return new PhpCompactor(
-            new DocblockAnnotationParser(
-                new DocblockParser(),
-                new AnnotationDumper(),
-                $ignoredAnnotations
-            )
-        );
+        return static function () use ($ignoredAnnotations): Compactor {
+            return new PhpCompactor(
+                new DocblockAnnotationParser(
+                    new DocblockParser(),
+                    new AnnotationDumper(),
+                    $ignoredAnnotations
+                )
+            );
+        };
     }
 
-    private static function createPhpScoperCompactor(stdClass $raw, string $basePath, ConfigurationLogger $logger): Compactor
+    private static function createPhpScoperCompactor(stdClass $raw, string $basePath, ConfigurationLogger $logger): Closure
     {
         $phpScoperConfig = self::retrievePhpScoperConfig($raw, $basePath, $logger);
 
-        $phpScoper = (new class() extends ApplicationFactory {
-            public static function createScoper(): Scoper
-            {
-                return parent::createScoper();
-            }
-        })::createScoper();
-
-        if ([] !== $phpScoperConfig->getWhitelistedFiles()) {
-            $whitelistedFiles = array_values(
-                array_unique(
-                    array_map(
-                        static function (string $path) use ($basePath): string {
-                            return make_path_relative($path, $basePath);
-                        },
-                        $phpScoperConfig->getWhitelistedFiles()
-                    )
+        $whitelistedFiles = array_values(
+            array_unique(
+                array_map(
+                    static function (string $path) use ($basePath): string {
+                        return make_path_relative($path, $basePath);
+                    },
+                    $phpScoperConfig->getWhitelistedFiles()
                 )
-            );
-
-            $phpScoper = new FileWhitelistScoper($phpScoper, ...$whitelistedFiles);
-        }
+            )
+        );
 
         $prefix = $phpScoperConfig->getPrefix() ?? unique_id('_HumbugBox');
 
-        return new PhpScoperCompactor(
-            new SimpleScoper(
-                $phpScoper,
-                $prefix,
-                $phpScoperConfig->getWhitelist(),
-                $phpScoperConfig->getPatchers()
-            )
-        );
+        return static function () use ($phpScoperConfig, $prefix, $whitelistedFiles): Compactor {
+            $phpScoper = (new class() extends ApplicationFactory {
```

Comment:
needs further refactoring in PhpScoper to avoid this. I think it's time to introduce a container in PhpScoper instead of those